cmd/car/detach.go

```diff
@@ -33,3 +35,35 @@ func DetachCar(c *cli.Context) error {
 	_, err = io.Copy(outStream, r.IndexReader())
 	return err
 }
+
+// DetachCarList prints a list of what's found in a detached index.
+func DetachCarList(c *cli.Context) error {
+	var err error
+
+	inStream := os.Stdin
+	if c.Args().Len() >= 1 {
+		inStream, err = os.Open(c.Args().First())
+		if err != nil {
+			return err
+		}
+		defer inStream.Close()
+	}
+
+	idx, err := index.ReadFrom(inStream)
+	if err != nil {
+		return err
+	}
+
+	if iidx, ok := idx.(index.IterableIndex); ok {
+		err := iidx.ForEach(func(mh multihash.Multihash, offset uint64) error {
+			fmt.Printf("%s %d\n", mh, offset)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return fmt.Errorf("index of codec %d is not iterable", idx.Codec())
```

Comment:
Would it be useful to print the codec name here instead?

```suggestion
	return fmt.Errorf("index of codec %s is not iterable", idx.Codec())
```

Reply:
can i just do `%T` to get a decent type name?

Reply:
I _think_ that'd print `multicodec.Code`.